wdl-lint/src/rules/matching_parameter_meta.rs

```diff
@@ -70,6 +70,29 @@ fn extra_param_meta(parent: &SectionParent, extra: &str, span: Span) -> Diagnost
     .with_fix("remove the extraneous key from the `parameter_meta` section")
 }
 
+/// Creates a "mismatched order" diagnostic.
+fn mismatched_param_order(parent: &SectionParent, span: Span, expected_order: &str) -> Diagnostic {
+    let (context, parent) = match parent {
+        SectionParent::Task(t) => ("task", t.name()),
+        SectionParent::Workflow(w) => ("workflow", w.name()),
+        SectionParent::Struct(s) => ("struct", s.name()),
+    };
+
+    Diagnostic::note(format!(
+        "parameter metadata in {context} `{parent}` is out of order",
+        parent = parent.text(),
+    ))
+    .with_rule(ID)
+    .with_label(
+        "parameter metadata must be in the same order as inputs",
+        span,
+    )
+    .with_fix(format!(
+        "order the parameter metadata as:\n{}",
+        expected_order
+    ))
+}
+
 /// Detects missing or extraneous entries in a `parameter_meta` section.
 #[derive(Default, Debug, Clone, Copy)]
 pub struct MatchingParameterMetaRule {
```

Comment:
Now that #371 is complete, this will need to implement `related_rules`. This should link to the `InputSorted` rule.